### PR TITLE
Alpha

### PR DESCRIPTION
**Pull request by safar1212**
In this pull request, I completed the app with the below features.

1: Splash screen

A simple page with the name of your app (yes, you need to choose one), and links to the signup and log-in pages.

2: Sign-up and login pages

3: The user should be able to register in the app with full name, email, and password (all mandatory).
The user can log into the app using email and password.
If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).
Home page (categories page)

4: When the user logs in, they are presented with the categories page.
For each category, the user can see their name, icon, and the total amount of all the transactions that belong to that category.
When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
There is a button "add a new category" at the bottom that brings the user to the page to create a new category.
Transactions page

5: For a given category, the list of transactions is presented, ordered by the most recent.
At the top of the page, the user can see the total amount for the category (the sum of all of the amounts of the transactions in that category).
There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
When the user clicks on the "Back" button (<), the user navigates to the home page.
"Add a new category" page

6: The user fills out a form to create a new category, indicating their name and icon (both mandatory).
The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
When the user clicks on the "Back" button (<), the user navigates to the home page.
"Add a new transaction" page

7: The user fills out a form to create a new transaction with:
name (mandatory)
amount (mandatory)
categories (mandatory at least one)
The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

The link to video

https://www.loom.com/share/caf54724114c40a5bfeed140aafb5f3f

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,81 @@
-# alpha-finance
\ No newline at end of file
+![](https://img.shields.io/badge/Microverse-blueviolet)
+
+
+# alpha-finance
+
```
Comment (SajjadAhmad14) [inline README.md:5]: The app should be deployed using Heroku, Netlify, or GitHub Pages and a link to the deployed project should be present in the README.

File: app/views/catagories/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<div style='background-color: #3778c2; height: 100vh;'>
+
+  <div class="pt-5">
+
+    <%= form_with model: @catagory, :html => {:id => "create_catagory"} do |f| %>
```
Comment (SajjadAhmad14) [inline app/views/catagories/new.html.erb:15]: The user is not able to add a new category on the `categories/new` page. Moreover, please add a back button that takes the user back to the home page.

File: app/models/catagory.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+class Catagory < ApplicationRecord
+  mount_uploader :icon, IconUploader
+
```
Comment (SajjadAhmad14) [inline app/models/catagory.rb:6]: Please validate user input in the `Item`, and `Catagory` models to make sure that anyone with bad intentions cannot compromise your app.

File: app/views/catagories/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,62 @@
+<div style='background-color: #dedede; height: 100vh;'>
+
+
```
Comment (SajjadAhmad14) [inline app/views/catagories/index.html.erb:3]: Please follow design guidelines for Typography and colors.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Safar,
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
Git flow is followed. :heavy_check_mark:
PR has a title and description. :heavy_check_mark:
## To Improve 🙏
- [ ] Please add integration tests for all the most important components of the app.
- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: app/views/catagories/index.html.erb (hunk 4 above)
Comment (safar1212) [reply app/views/catagories/index.html.erb:3]: Thanks for the recommendations @SajjadAhmad14 .
Actually, for the colors, I am using the colors required in the project.
I don't think there is problem with colors

File: app/views/catagories/new.html.erb (hunk 2 above)
Comment (safar1212) [reply app/views/catagories/new.html.erb:15]: Thank you for your recommendation. @SajjadAhmad14 
As I checked I can add a new category without any issues. Could you please tell me exactly where the issue is?

File: app/models/catagory.rb (hunk 3 above)
Comment (zainsadaqat) [inline app/models/catagory.rb:6]: As requested by the Previous Code Reviewer, Please validate user input in the Item, and Catagory models to make sure that anyone with bad intentions cannot compromise your app. 👍🏻

**Review by zainsadaqat — CHANGES_REQUESTED** [review]
# Required Changes 🔁 
# Hello @safar1212  😃 

You're doing a great job on your project. We're proud of you. Good job so far! 

There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
  ✅  Correct use of Github Flow
  ✅  No linter errors
  ✅  Descriptive Pull Request

## Required Changes ♻️

_Check the comments under the review._




## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding! 👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 5]
```diff
@@ -1 +1,85 @@
...
+|------|--------|---------|----------|
+|Safar Ali|[@Safar Ali](https://github.com/safar1212)|[@Safar Ali](https://twitter.com/SafarAli999)|[@Safar Ali](https://www.linkedin.com/in/safar-ali999/)|
+
+## 🤝 Contributing
+
+Contributions, issues, and feature requests are welcome!
+
+## Aknowledgements
+
+Original design idea by Gregoire Vella on Behance
```
Comment (muneebulrehman) [inline README.md:76]: - Optional
Kindly give the credit to the original designer of the design you have used. I am raising this as optional as the previous reviewer didn't raise this as required.
image

**Review by muneebulrehman — APPROVED** [review]
Hi @safar1212 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🥳
Congratulations! 🎉 

# Status: Approved✔️

## Highlights
- No linter errors
- Descriptive PR
- Good follow-up with the previous reviewer.

## Optional suggestions

- [ ] The live version of your app is breaking. Kindly fix this error. I am raising this as the optional since the previous reviewer didn't raise this as required.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 5 above)
Comment (safar1212) [reply README.md:76]: Oho! Thanks for reminding me of that. I actually forgot that. I will surely do it.

Comment (safar1212) [reply README.md:76]: @muneebulrehman I am not able to submit the project on the dashboard. I think you've forgotten to toggle that option. Could you please change that? I am still getting the "Review pending" message.

Comment (muneebulrehman) [reply README.md:76]: @safar1212, I submitted it just after I left the comment here. Are you still having this issue?

**Pull request merged**